.github/DeliverToNuGet.ps1

```diff
@@ -0,0 +1,124 @@
+)
+
+function GenerateNuspec
+(
+    [Parameter(Mandatory=$true)]
+    $PackageId,
+    [Parameter(Mandatory=$true)]
+    $Version,
+    [Parameter(Mandatory=$true)]
+    $Authors,
+    [Parameter(Mandatory=$true)]
+    $Owners
+)
+{
+    [xml] $template = Get-Content "$PSScriptRoot\ALAppExtensions.nuspec"
+
+    $template.package.metadata.id = $PackageId
+    $template.package.metadata.version = $Version
+    $template.package.metadata.authors = $Authors
+    $template.package.metadata.owners = $Owners
+
+    return $template
+}
+
+try {
+    $nuGetAccount = $parameters.Context | ConvertFrom-Json | ConvertTo-HashTable
+    $nuGetServerUrl = $nuGetAccount.ServerUrl
+    $nuGetToken = $nuGetAccount.Token
+}
+catch {
+    throw "NuGetContext secret is malformed. Needs to be formatted as JSON, containing serverUrl and token."
+}
+
+if (-not ($nuGetServerUrl)) {
+    throw "Cannot retrieve NuGet server URL from NuGetContext"
+} 
+
+if (-not $nuGetToken) {
+    throw "Cannot retrieve NuGet token  URL from NuGetContext"
+} 
+
+Write-Host "Successfully retrieved information from NuGetContext" -ForegroundColor Green
+
+$project = $parameters.project
+$projectName = $parameters.projectName
+$appsFolder = $parameters.appsFolder
+$testAppsFolder = $parameters.testAppsFolder
+$type = $parameters.type
+
+# Construct package ID
+$packageId = "$($env:GITHUB_REPOSITORY_OWNER)-$($env:RepoName)"
+
+if (-not $project -or ($project -ne '.')) {
+    $packageId += "-$projectName"
+}
+
+if ($type -eq 'CD') {
+    $packageId += "-preview"
+}
+
+# Extract version from the published folders (naming convention)
+$packageVersion = $appsFolder.SubString($appsFolder.IndexOf("-Apps-") + "-Apps-".Length) #version is right after '-Apps-'
+
+$nuspec = GenerateNuspec `
+            -PackageId $packageId `
+            -Version $packageVersion `
+            -Authors "$env:GITHUB_REPOSITORY_OWNER" `
+            -Owners "$env:GITHUB_REPOSITORY_OWNER"
+
+# Create a temp folder to use for the packaging
+$packageFolder = Join-Path $env:TEMP ([GUID]::NewGuid().ToString())
+New-Item -Path $packageFolder -ItemType Directory | Out-Null
+
+Write-Host "Package folder: $packageFolder" -ForegroundColor Magenta
+
+try {
+    $outputDirectory = Join-Path $env:GITHUB_WORKSPACE 'out'
+    New-Item -Path $outputDirectory -ItemType Directory -Force | Out-Null
+
+    $appsPackage = Join-Path $appsFolder 'Package'
+    if(Test-Path -Path "$appsPackage") {
+        Copy-Item -Path "$appsPackage" -Destination "$packageFolder/Apps/" -Recurse -Container -Force 
+    }
+
+    $testAppsPackage = Join-Path $testAppsFolder 'Package'
+    if(Test-Path -Path "$testAppsPackage") {
+        Copy-Item -Path "$testAppsPackage" -Destination "$packageFolder/Tests/" -Recurse -Container -Force
+    }
+    
+    #Create .nuspec file
+    $nuspecFilePath = (Join-Path $packageFolder 'manifest.nuspec')
+    $nuspec.Save($nuspecFilePath)
+
+    Write-Host "Download nuget CLI" -ForegroundColor Magenta
```

Comment:
Should we move the download of Nuget.exe into a separate action instead, and reference that one? We might need it in more places?

Reply:
This is the part where AL-Go is a bit limited.
YAML files are only to be edited from AL-Go itself.

I see this as an obstacle, because as much as we try, we might not be able to have "one solution fits them all" scenario.